### ISSUE TITLE
[feat]: provide intellisense for `package.json` configs

### ISSUE THREAD

**Issue by helmturner** [absolute-version/commit-and-tag-version#67]
This is more of a nice-to-have since the conversion to TypeScript will allow for using a strongly typed config file. For this, we would need to generate JSON schemas to represent the configuration options.

Within the month I'll be releasing a library, inspired by [zod](https://github.com/colinhacks/zod), which has a similar API but can also generate JSON schemas. This would kill two birds: we get strong validation for configuration options at run-time and can use the same validators to generate the JSON schema during CI (for static validation).

Basically, I'm just parking this thought as a reminder to myself 😄
